mongo/options/aggregateoptions.go

```diff
@@ -8,81 +8,99 @@ package options
 
 import "time"
 
-// AggregateOptions represents all possible options to the Aggregate() function.
+// AggregateOptions represents options that can be used to configure an Aggregate operation.
 type AggregateOptions struct {
-	AllowDiskUse             *bool          // Enables writing to temporary files. When set to true, aggregation stages can write data to the _tmp subdirectory in the dbPath directory
-	BatchSize                *int32         // The number of documents to return per batch
-	BypassDocumentValidation *bool          // If true, allows the write to opt-out of document level validation. This only applies when the $out stage is specified
-	Collation                *Collation     // Specifies a collation
-	MaxTime                  *time.Duration // The maximum amount of time to allow the query to run
-	MaxAwaitTime             *time.Duration // The maximum amount of time for the server to wait on new documents to satisfy a tailable cursor query
-	Comment                  *string        // Enables users to specify an arbitrary string to help trace the operation through the database profiler, currentOp and logs.
-	Hint                     interface{}    // The index to use for the aggregation. The hint does not apply to $lookup and $graphLookup stages
+	// If true, the operation can write to temporary files in the _tmp subdirectory of the database directory path on
+	// the server. The default value is false.
+	AllowDiskUse *bool
+
+	// The maximum number of documents to be included in each batch returned by the server.
+	BatchSize *int32
+
+	// If true, writes executed as part of the operation will opt out of document-level validation on the server. This
+	// option is valid for MongoDB versions >= 3.2 and is ignored for previous server versions. The default value is
+	// false. See https://docs.mongodb.com/manual/core/schema-validation/ for more information about document
+	// validation.
+	BypassDocumentValidation *bool
+
+	// Specifies a collation to use for string comparisons during the operation. This option is only valid for MongoDB
+	// versions >= 3.4. For previous server versions, the driver will return an error if this option is used. The
+	// default value is nil, which means the default collation of the collection will be used.
+	Collation *Collation
+
+	// The maximum amount of time that the query can run on the server. The default value is nil, meaning that there
+	// is no time limit for query execution.
+	MaxTime *time.Duration
+
+	// The maximum amount of time that the server should wait for new documents to satisfy a tailalbe cursor query.
+	// This option is only valid for MongoDB versions >= 3.2 and is ignored for previous server versions.
```

Comment:
Did you consider mentioning which stage (implicitly) creates a tailable cursor?  I think it's just $changeStream currently.

Reply:
I'd prefer to not mention this. Someone using `$changeStream` manually likely already knows that they're creating a change stream rather than a normal cursor, so I don't think it matters much there.